README.md

```diff
@@ -1,247 +1,220 @@
-      or [Defaults.MEASUREMENT_FORKS (`5`)](https://javadoc.io/static/org.openjdk.jmh/jmh-core/1.21/org/openjdk/jmh/runner/Defaults.html#MEASUREMENT_FORKS) if it is not specified by `@Fork`.
-  * (Kotlin/Js and Wasm) `jsUseBridge` – when `false` disables to generate special benchmark bridges to prevent inlining optimisations (only for `BuiltIn` benchmark executors).
-
-Time units can be NANOSECONDS, MICROSECONDS, MILLISECONDS, SECONDS, MINUTES, or their short variants such as "ms" or "ns".  
-  
-Example: 
-
-```groovy
-benchmark {
-    // Create configurations
-    configurations {
-        main { // main configuration is created automatically, but you can change its defaults
-            warmups = 20 // number of warmup iterations
-            iterations = 10 // number of iterations
-            iterationTime = 3 // time in seconds per iteration
-        }
-        smoke {
-            warmups = 5 // number of warmup iterations
-            iterations = 3 // number of iterations
-            iterationTime = 500 // time in seconds per iteration
-            iterationTimeUnit = "ms" // time unit for iterationTime, default is seconds
-        }   
+4.  **Using AllOpen Plugin**: The `allopen` plugin is used to satisfy JMH requirements, or all benchmark classes and methods should be `open`.
+
+    ```
+    allOpen {
+        annotation("org.openjdk.jmh.annotations.State")
     }
-    
-    // Setup targets
-    targets {
-        // This one matches compilation base name, e.g. 'jvm', 'jvmTest', etc
-        register("jvm") {
-            jmhVersion = "1.21" // available only for JVM compilations & Java source sets
-        }
-        register("js") {
-            // Note, that benchmarks.js uses a different approach of minTime & maxTime and run benchmarks
-            // until results are stable. We estimate minTime as iterationTime and maxTime as iterationTime*iterations
-            //
-            // You can configure benchmark executor - benchmarkJs or buildIn (works only for JsIr backend) with the next line:
-            // jsBenchmarksExecutor = JsBenchmarksExecutor.BuiltIn
+    ```
+
+5.  **Specifying Repository**: Ensure you have `mavenCentral()` in the list of repositories:
+
+        ```groovy
+        repositories {
+            mavenCentral()
         }
-        register("native")
-        register("wasm") // Experimental
-    }
-}
-```  
-  
-# Separate source sets for benchmarks
+        ```
+
+    </details>
+
+### Target-specific configurations
 
-Often you want to have benchmarks in the same project, but separated from main code, much like tests. Here is how:
+#### Kotlin/JS
 
-Define source set:
-```groovy
-sourceSets {
-    benchmarks
+For Kotlin/JS, include the `nodejs()` method call in the `kotlin` block:
```

Comment:
For benchmarking Kotlin/JS code Node.js execution enviroment should be targeted. See https://kotlinlang.org/docs/js-project-setup.html#execution-environments

This is because kotlinx-benchmark-runtime uses Node.js environment to run benchmarks.
By default Benchmark.js library (https://benchmarkjs.com) is used under the hood, see `JsBenchmarkExecutor.kt` file

Reply:
I appreciate the resources and feedback. I have updated the kotlin/js section of the readme with this info [22e886a](https://github.com/Kotlin/kotlinx-benchmark/pull/117/commits/22e886a9380be053721184ffd87d450175d6a51c)